test_db::test_pooloverload fails with Python3.10

### Description

Running test_db::test_pooloverload raises the following errors in a Python 3.10 test environment:

```
=================================== FAILURES ===================================
______________________________ test_pooloverload _______________________________
[gw0] linux -- Python 3.10.4 /opt/venv/bin/python
test_db.py:55: in test_pooloverload
    patch_flask = patch('flask.request', spec=flaskmock)
/usr/local/lib/python3.10/unittest/mock.py:1766: in patch
    return _patch(
/usr/local/lib/python3.10/unittest/mock.py:1287: in __init__
    raise InvalidSpecError(
E   unittest.mock.InvalidSpecError: Cannot spec attr 'request' as the spec has already been mocked out. [spec=<Mock id='139869859266080'>]
=============================== warnings summary ===============================
```

### Steps to reproduce

https://github.com/maany/rucio/actions/runs/3676199993/jobs/6216561082 

### Rucio Version

_No response_

### Additional Information

_No response_

--- a/lib/rucio/tests/test_db.py
+++ b/lib/rucio/tests/test_db.py
@@ -14,7 +14,7 @@
 # limitations under the License.
 
 import pytest
-from unittest.mock import patch, Mock
+from unittest.mock import patch
 
 from rucio.db.sqla.session import get_session, _get_engine_poolclass, NullPool, QueuePool, SingletonThreadPool
 from rucio.common.exception import InputValidationError
@@ -48,11 +48,12 @@ def test_pooloverload():
     # Create a new ErrorHandlingMethodView as_view
     ping_view = Ping.as_view('ping')
 
-    # need to specify the patch more closely here
-    # for python >= 3.8, which leads to errors
-    flaskmock = Mock()
-    flaskmock.method = 'replacement string!'
-    patch_flask = patch('flask.request', spec=flaskmock)
+    # specification for the mock we create to replace flask.request
+    # without specifying this, _is_async_obj is run which triggers flask RuntimeError
+    class T:
+        method = 'replacement string'
+
+    patch_flask = patch('flask.request', spec=T)
 
     patch_getheaders = patch('rucio.web.rest.flaskapi.v1.ping.Ping.get_headers')
     patch_dispatch = patch(

